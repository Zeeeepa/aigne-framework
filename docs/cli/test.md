# Testing

Use AIGNE CLI to run agent tests.

## `aigne test` Command

Run tests in the specified agent directory.

```bash
aigne test [options]
```

### Options

* `--path <path>`: Path to the agent directory (defaults to current directory `.`)
* `--help`: Show command help

### Examples

#### Run tests in current directory

```bash
aigne test
```

#### Run tests in specific directory

```bash
aigne test --path ./my-agents
```

### Test File Structure

The AIGNE testing system automatically finds and runs test files in your project. Make sure your test files follow the correct naming conventions and structure.

### Best Practices

1. **Organize test files**: Place test files in appropriate directory structures
2. **Write clear test cases**: Ensure each test case has a clear purpose and expected outcome
3. **Use descriptive test names**: Make test names clearly express what the test covers
4. **Run tests regularly**: Run tests regularly during development to ensure code quality
