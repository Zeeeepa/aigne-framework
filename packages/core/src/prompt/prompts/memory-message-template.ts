export const MEMORY_MESSAGE_TEMPLATE = `\
Here are some useful memories to consider:

<memories>
{{memories}}
</memories>
`;
