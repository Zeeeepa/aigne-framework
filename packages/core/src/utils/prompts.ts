function buildJsonPrompt(schema: Record<string, unknown> | string, firstLine: string) {
  let prompt = firstLine;
  if (typeof schema === "string") {
    prompt += `\n<json_fields>\n${schema}\n</json_fields>`;
  } else {
    prompt += `\n<json_fields>\n${JSON.stringify(schema)}\n</json_fields>`;
  }
  prompt +=
    "\nDo not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON.";
  return prompt;
}

export function getJsonOutputPrompt(schema: Record<string, unknown> | string) {
  return buildJsonPrompt(
    schema,
    "Output must be a JSON object containing the following fields only.",
  );
}

export function getJsonToolInputPrompt(schema: Record<string, unknown> | string) {
  return buildJsonPrompt(
    schema,
    "Tool input must be a JSON object containing the following fields only.",
  );
}
