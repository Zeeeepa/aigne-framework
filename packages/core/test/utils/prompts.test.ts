import { describe, expect, test } from "bun:test";
import { getJsonOutputPrompt, getJsonToolInputPrompt } from "../../src/utils/prompts.js";

describe("getJsonOutputPrompt", () => {
  test("string schema", () => {
    const schema = "{name: string, age: number}";
    expect(getJsonOutputPrompt(schema)).toMatchSnapshot();
  });

  test("object schema", () => {
    const schema = { name: "string", age: "number" };
    expect(getJsonOutputPrompt(schema)).toMatchSnapshot();
  });

  test("should handle empty schema", () => {
    const schema = {};
    expect(getJsonOutputPrompt(schema)).toMatchSnapshot();
  });

  test("should handle empty string schema", () => {
    const schema = "";
    expect(getJsonOutputPrompt(schema)).toMatchSnapshot();
  });
});

describe("getJsonToolInputPrompt", () => {
  test("string schema", () => {
    const schema = "{foo: string}";
    expect(getJsonToolInputPrompt(schema)).toMatchSnapshot();
  });

  test("object schema", () => {
    const schema = { foo: "string" };
    expect(getJsonToolInputPrompt(schema)).toMatchSnapshot();
  });
});
