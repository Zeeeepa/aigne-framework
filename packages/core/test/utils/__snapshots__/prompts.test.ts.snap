// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`getJsonOutputPrompt string schema 1`] = `
"Output must be a JSON object containing the following fields only.
<json_fields>
{name: string, age: number}
</json_fields>
Do not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON."
`;

exports[`getJsonOutputPrompt object schema 1`] = `
"Output must be a JSON object containing the following fields only.
<json_fields>
{"name":"string","age":"number"}
</json_fields>
Do not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON."
`;

exports[`getJsonOutputPrompt should handle empty schema 1`] = `
"Output must be a JSON object containing the following fields only.
<json_fields>
{}
</json_fields>
Do not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON."
`;

exports[`getJsonOutputPrompt should handle empty string schema 1`] = `
"Output must be a JSON object containing the following fields only.
<json_fields>

</json_fields>
Do not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON."
`;

exports[`getJsonToolInputPrompt string schema 1`] = `
"Tool input must be a JSON object containing the following fields only.
<json_fields>
{foo: string}
</json_fields>
Do not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON."
`;

exports[`getJsonToolInputPrompt object schema 1`] = `
"Tool input must be a JSON object containing the following fields only.
<json_fields>
{"foo":"string"}
</json_fields>
Do not include any explanations, comments, or Markdown formatting (such as triple backticks). Return only the raw JSON."
`;
